### ISSUE TITLE
Large chunks of my schema is missing in the openapi output.

### ISSUE THREAD

**Issue by FelixZY** [blomqma/next-rest-framework#39]
I have a mildly complex schema called `ProfileReferenceDTOSchema`. Unfortunately, the generated openapi file does not include the full schema:

![Screenshot from 2023-05-07 07-30-14](https://user-images.githubusercontent.com/2656517/236659760-99e6bd9a-e501-4fca-a519-69a40a47d1a9.png)

* Left: Rendered by `next-rest-framework`
* Middle: `output` declaration in code
* Right: Rendered by `@asteasolutions/zod-to-openapi`

JSON output

```json
{
  "200": {
    "content": {
      "application/json": {
        "schema": {
          "type": "array",
          "items": {
            "type": "object",
            "properties": {
              "id": { "type": "string" },
              "name": { "type": "string" },
              "images": { "type": "object", "properties": {} }
            }
          }
        }
      }
    }
  }
}
```

I'm not sure why this is but it breaks our documentation.

**Comment by blomqma** [comment]
Hey, is the `ProfileReferenceDTOSchema` something you could share to let me better debug your issue?

**Comment by FelixZY** [comment]
Fortunately, yes. Here it is:

```typescript
import { extendZodWithOpenApi } from "@asteasolutions/zod-to-openapi";
import { ProfileType } from "@prisma/client";
import z from "zod";

extendZodWithOpenApi(z);

type NamedReferenceDTO = z.infer<typeof NamedReferenceDTOSchema>;
const NamedReferenceDTOSchema = z.object({
  id: z.string().uuid(),
  name: z.string(),
});

export type ImageDTO = z.infer<typeof ImageDTOSchema>;
export const ImageDTOSchema = z.object({
  cover: z.nullable(z.string().url()).openapi({
    description: "...",
    example: "https://...",
  }),
  poster: z.nullable(z.string().url()).openapi({
    description: "...",
    example: "https://...",
  }),
  square: z.nullable(z.string().url()).openapi({
    description: "...",
    example: "https://...",
  }),
});

export type ProfileReferenceDTO = z.infer<typeof ProfileReferenceDTOSchema>;
export const ProfileReferenceDTOSchema = NamedReferenceDTOSchema.merge(
  z.object({
    type: z.enum([
      ProfileType.organization,
      ProfileType.individual,
      ProfileType.venue,
    ]),
    images: ImageDTOSchema,
  })
);

type ProfileDTO = z.infer<typeof ProfileDTOSchema>;
const ProfileDTOSchema = z.intersection(
  NamedReferenceDTOSchema.merge(
    z.object({
      images: ImageDTOSchema,
      description: z.string(),
      links: z.array(
        z.object({
          url: z.string().url(),
        })
      ),
    })
  ),
  z.discriminatedUnion("type", [
    z.object({
      type: z.literal(ProfileType.organization),
      members: z.array(ProfileReferenceDTOSchema),
    }),
    z.object({
      type: z.literal(ProfileType.individual),
      organizations: z.array(ProfileReferenceDTOSchema),
    }),
    z.object({
      type: z.literal(ProfileType.venue),
      coords: z.object({
        lat: z.number(),
        lng: z.number(),
      }),
      parentVenue: z.nullable(ProfileReferenceDTOSchema),
      childVenues: z.array(ProfileReferenceDTOSchema),
    }),
  ])
);

```

**Comment by blomqma** [comment]
Thanks, I was able to reproduce the issue with your example & create a fix for it. There was a bug where `intersection`, `enum` and `nullable` Zod types were not supported by the OpenAPI spec generation and those are now fixed in the latest [v0.7.1](https://github.com/blomqma/next-rest-framework/releases/tag/v0.7.1) release. I need to look deeper into the Zod types and see if there are any other types that are not yet supported by the framework but I think this should address your issue for now.
